plugins/outputs/cloudwatchlogs/pusher.go

```diff
@@ -330,23 +330,36 @@ func (p *pusher) createLogGroupAndStream() error {
 		LogStreamName: &p.Stream,
 	})
 
-	if err != nil {
-		p.Log.Debugf("creating stream fail due to : %v \n", err)
-		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
-			_, err = p.Service.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
-				LogGroupName: &p.Group,
-			})
+	if err == nil {
+		p.Log.Debugf("successfully created log stream %v", p.Stream)
+		return nil
+	}
 
-			// create stream again if group created successfully.
+	p.Log.Debugf("creating stream fail due to : %v", err)
+	if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
+		_, err = p.Service.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
+			LogGroupName: &p.Group,
+		})
+
+		// attempt to create stream again if group created successfully.
+		if err == nil {
+			p.Log.Debugf("successfully created log group %v. Retrying log stream %v", p.Group, p.Stream)
+			_, err = p.Service.CreateLogStream(&cloudwatchlogs.CreateLogStreamInput{
+				LogGroupName:  &p.Group,
+				LogStreamName: &p.Stream,
+			})
 			if err == nil {
-				_, err = p.Service.CreateLogStream(&cloudwatchlogs.CreateLogStreamInput{
-					LogGroupName:  &p.Group,
-					LogStreamName: &p.Stream,
-				})
-			} else {
-				p.Log.Errorf("creating group fail due to : %v \n", err)
+				p.Log.Debugf("successfully created log stream %v", p.Stream)
 			}
+		} else {
+			p.Log.Debugf("creating group fail due to : %v", err)
 		}
+
+	}
+
+	if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == cloudwatchlogs.ErrCodeResourceAlreadyExistsException {
```

Comment:
IMO, please put handling error `ErrCodeResourceAlreadyExistsException` before [`ErrCodeResourceNotFoundException`](https://github.com/aws/amazon-cloudwatch-agent/pull/416/files#diff-9a19b49be59e15932ccf98dbea45ca58f52d90639bbfef76e2cbc93dc646cb88R339)

Reply:
Doing that is a bigger change. The existing code reuses the `err` variable, and attempts the following:
1. Try to create log stream
2. If it fails, check if it's because the log group doesn't exist yet (ResourceNotFound)
3. Then, try to create the log group
4. If it succeeds, then try to create the log stream again
I didn't change this flow.

Trying to check for `ResourceAlreadyExists` first would change it to:
1. Try to create log stream
2. If it fails because it already exist, return nil
3. Else, if it fails because the log group doesn't exist yet
4. Then try to create the log group
5. If it fails because it already exists, OR the log group was created successfully
6. Then try to create the log stream again

5 in the second scenario is why I didn't update the order to begin with, opting to keep the existing flow so when it fails through to this check, we can catch the error gracefully at the very end.

Reply:
I see. I have not notice that you reuse the `err` variable for the second create `Log Stream`, which is to avoid the 5 condition. Agree with this changes.